src/reducers/clusterReducer.js

```diff
@@ -75,14 +30,24 @@ export default function clusterReducer(
 
   switch (action.type) {
     case types.CLUSTERS_LOAD_SUCCESS:
-      items = {};
+      // if state was populated previously, let new data overwrite old data partially
+      var prevClusterIDs = Object.keys(state.items).sort();
+
+      // use existing state's items and update it
+      items = Object.assign(state.items, action.clusters);
```

Comment:
```suggestion
      items = Object.assign({}, state.items, action.clusters);
```

So that we get a new object, instead of mutating state.items.

I'm actually a bit surprised it didn't trigger a warning or error from redux. My understanding is a bit shaken. I'll check out the branch and do some poking tomorrow ok?

Reply:
sure